CHANGES.rst

```diff
@@ -506,6 +506,9 @@ astropy.wcs
 
 - Fixed a crash while loading a WCS from headers containing duplicate SIP keywords. [#8893]
 
+- Fixed a possible buffer overflow when using too large negative indices for
+  ``cunit`` or ``ctype`` [#9151]
```

Comment:
Any negative indices, right? (not just too large ones?)

Reply:
No, negative indices `< -size` are allowed and work.

The `sq_ass_item` gets the index that is already wrapped around so the methods where I fixed that get a positive index as long as the user-provided index is within `-size <= index < size` (see [`PySequenceMethods.sq_ass_item`](https://docs.python.org/3/c-api/typeobj.html#c.PySequenceMethods.sq_ass_item)).

Reply:
Ok perfect, thanks!